tests/integration/kubernetes/gha-run.sh

```diff
@@ -47,8 +47,14 @@ function create_cluster() {
 }
 
 function install_bats() {
-    sudo apt-get update
-    sudo apt-get -y install bats
+    BATS_REPO="https://github.com/bats-core/bats-core.git"
+
+    echo "Install BATS from sources"
+    pushd "${GOPATH}/src/"
+    git clone "${BATS_REPO}" || true
```

Comment:
What happens if `bats` has a bug that blocks us?  Who'll take care of debugging and downgrading this?

I'd like to fuly understand, preferrably with one example, what's exactly needed and how those new setup_file nd teardown_file will be used, to make sure taking this approach is worth the risk.

Moreover, we could add this to versions.yaml and pick it up from there, but again I'm looking at the maintenance side of things and I wonder who'll be responsible to:
1. Check that new releases of bats are available
2. Test that new releases can be updated here
3. Make sure to open PRs once new releases are released

Reply:
It seems we can do something like:
* Install `add-apt-repository`
  * `apt install -y software-properties-common`
* Enable lunar's universe repo
  * `add-apt-repository 'deb http://archive.ubuntu.com/ubuntu/ lunar universe'
* Install bats from there
  * `apt install -y bats`
* Remove the lunar's repo
  * `add-apt-repository --remove 'deb http://archive.ubuntu.com/ubuntu/ lunar universe'

This makes me way more confident, as the distro theoretically will test the package, than just installing one specific version, or counting on `main` to be always working.

@UnmeshDeodhar, WDYT?

Reply:
Sounds good to me. Let me verify the version it installs.

Reply:
This installs `Bats 1.8.2` so I am just going to change my PR and add this. Thanks @fidencio !